test/suites/greenboot.robot

```diff
@@ -0,0 +1,142 @@
+
+    [Teardown]    Run Keywords
+    ...    Cleanup User Workload
+    ...    Cleanup And Start
+
+Simulate Service Failure
+    [Documentation]    Simulate Service failure
+    Restart Greenboot And Wait For Success
+    Disrupt Service
+    Cleanup MicroShift    "--keep-images"
+    Run Keyword And Expect Error    0 != 1
+    ...    Systemctl    enable    --now microshift
+    Run Keyword And Expect Error    0 != 1
+    ...    Restart Greenboot And Wait For Success
+
+    [Teardown]    Run Keywords
+    ...    Restore Service
+    ...    Cleanup And Start
+
+Simulate Pod Failure
+    [Documentation]    Simulate pod network failure
+    Restart Greenboot And Wait For Success
+    Disrupt Pod Network
+    Restart MicroShift
+    Run Keyword And Expect Error    0 != 1
+    ...    Restart Greenboot And Wait For Success
+
+    [Teardown]    Run Keywords
+    ...    Restore Default MicroShift Config
+    ...    Cleanup And Start
+
+
+*** Keywords ***
+Setup Suite
+    [Documentation]    Set up all of the tests in this suite
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig
+    Upload String To File    ${GREENBOOT_CONFIG_CONTENT}    ${GREENBOOT_CONFIG_FILE} # change the default timeout
+
+Add User Workload
+    [Documentation]    Upload User workload files to the MicroShift host
+    Put File    ${LOCAL_WORKLOAD_FILE}    /tmp/busybox_running_check.sh
+    ${stdout}    ${rc}=    Execute Command
+    ...    [ -f ${GREENBOOT_WORKLOAD_FILE} ] || sudo mv /tmp/busybox_running_check.sh ${GREENBOOT_WORKLOAD_FILE}
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+    Put Directory    ${MANIFEST_SOURCE_DIR}    /tmp/manifests
+    ${stdout}    ${rc}=    Execute Command
+    ...    cp -f /tmp/manifests/* ${MANIFEST_DIR} && sudo rm -rf /tmp/manifests
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+Cleanup User Workload
+    [Documentation]    Remove User workload files from the MicroShift host
+    ${stdout}    ${rc}=    Execute Command
+    ...    rm -rf ${GREENBOOT_WORKLOAD_FILE} ${MANIFEST_DIR}/*
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+Restart Greenboot And Wait For Success
+    [Documentation]    Restart the greenboot-healthcheck service and check its status
+    Systemctl    restart    greenboot-healthcheck.service
+    Wait Until Greenboot Health Check Exited
+
+Disrupt Service
+    [Documentation]    Prevent Microshift service from starting correctly.
+    ${is_ostree}=    Is System OSTree
+
+    ${stdout}    ${rc}=    Execute Command
+    ...    which hostname
+    ...    sudo=False    return_rc=True
+    IF    ${rc} == 0    Set Suite Variable    \${HOSTNAME_BIN_PATH}    ${stdout}
+
+    ${cmd}=    Set Variable    sudo chmod 000 ${HOSTNAME_BIN_PATH}
+    IF    ${is_ostree}
+        ${cmd}=    Set Variable    sudo rpm-ostree usroverlay && sudo chmod 000 $(which hostname)
```

Comment:
Sorry, just noticed: you're using both `${HOSTNAME_BIN_PATH}` and `$(which hostname)`. Could we use `${HOSTNAME_BIN_PATH}` in both cases or are there any obstacles?

Reply:
Yeap good catch :)